### PR TITLE
Fix panic in getNpdPodStat

### PR DESCRIPTION
**Pull request by dims**
Add some defensive code around gathering the cpuUsage, rss, workingSet information in the NPD related test case. 

Found in https://testgrid.k8s.io/google-gce#gci-gce-alpha-enabled-default&width=20

```
[PANICKED] Test Panicked
In [It] at: runtime/panic.go:261 @ 02/28/24 12:56:48.27

runtime error: invalid memory address or nil pointer dereference

Full Stack Trace
  k8s.io/kubernetes/test/e2e/node.getNpdPodStat({0x7f4fc429cc80?, 0xc002a88030?}, 0xc001bb0b90?, {0xc002d28180?, 0x0?})
  	k8s.io/kubernetes/test/e2e/node/node_problem_detector.go:387 +0x10b
  k8s.io/kubernetes/test/e2e/node.init.func7.2({0x7f4fc429cc80, 0xc002a88030})
  	k8s.io/kubernetes/test/e2e/node/node_problem_detector.go:206 +0x29a5
< Exit [It] should run without error - k8s.io/kubernetes/test/e2e/node/node_problem_detector.go:63 @ 02/28/24 12:56:48.27 (5.233s)
```

#### What type of PR is this?
/kind bug
/kind failing-test

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/node/node_problem_detector.go [hunk 1]
```diff
@@ -384,9 +384,13 @@ func getNpdPodStat(ctx context.Context, f *framework.Framework, nodeName string)
 		if !strings.HasPrefix(pod.PodRef.Name, "node-problem-detector") {
 			continue
 		}
-		cpuUsage = float64(*pod.CPU.UsageNanoCores) * 1e-9
-		rss = float64(*pod.Memory.RSSBytes) / 1024 / 1024
-		workingSet = float64(*pod.Memory.WorkingSetBytes) / 1024 / 1024
+		if pod.CPU != nil {
```
Comment (kannon92) [inline test/e2e/node/node_problem_detector.go:387]:
```suggestion
		```suggestion
		if pod.CPU.UsageNanoCores != nil {
```
Technically you still could get a panic if UsageNanoCores was undefined.
```

File: test/e2e/node/node_problem_detector.go [hunk 2]
```diff
@@ -384,9 +384,13 @@ func getNpdPodStat(ctx context.Context, f *framework.Framework, nodeName string)
...
+			cpuUsage = float64(*pod.CPU.UsageNanoCores) * 1e-9
+		}
+		if pod.Memory != nil {
```
Comment (kannon92) [inline test/e2e/node/node_problem_detector.go:390]:
```suggestion
		if pod.Memory.RSSBytes != nil && pod.Memory.WorkingSetBytes != nil {
```

File: test/e2e/node/node_problem_detector.go (hunk 1 above)
Comment (dims) [reply test/e2e/node/node_problem_detector.go]: you are right. i gotta check nil on both as well.

File: test/e2e/node/node_problem_detector.go (hunk 2 above)
Comment (dims) [reply test/e2e/node/node_problem_detector.go:390]: same as above

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**